### PR TITLE
Refactor things that run each tick into nicer tickable service interface

### PR DESCRIPTION
**Pull request by TraceLD**

### CODE REVIEW

**Comment by john01dav** [comment]
Overall, this is a very good change, but I see two concerns:

- In Java, IInterface isn't the standard, so ITickableServic should be renamed. 
- This might be over-engineering, but it would be cleaner and more reusable to refactor the every x ticks logic from the broadcast service into an abstract class that implements TickableService. Said class would have a final implementation of tick() and an abstract method with a name like onRun() or something.

**Review by john01dav — CHANGES_REQUESTED** [review]
See my previous comment (it didn't register as a review in Github for some reason).

File: server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java [hunk 1]
```diff
@@ -6,39 +6,35 @@
...
     private final String id;
     private final String prefix;
     private final List<String> broadcastMessages;
     private final Server server;
 
-    private int broadcastsTimer;
     private int broadcastMsgsIndex;
 
-    public static final int BROADCAST_EVERY_X_IN_TICKS = (10 * 60 * 20);
+    public static final int INTERVAL_IN_TICKS = (10 * 60 * 20);
```
Comment (john01dav) [inline server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java:17]: While we're here, there should probably be a comment saying that this is 10 mintues (at least, I think that it is, hence the need for a comment).

File: server/src/main/java/club/moddedminecraft/polychat/server/services/TimedTickableService.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        } else {
+            timer += 1;
+        }
+    }
+
+
+    /**
+     *  Triggered every {@link TimedTickableService#intervalInTicks} ticks.
+     */
+    public void onRun() {}
```
Comment (john01dav) [inline server/src/main/java/club/moddedminecraft/polychat/server/services/TimedTickableService.java]: This should be `public abstract void onRun();`.

**Review by john01dav — CHANGES_REQUESTED** [review]

File: server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java (hunk 1 above)
Comment (TraceLD) [reply server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java:17]: I honestly have no clue. This interval was agreed on months ago and I don't remember.

**Review by john01dav — APPROVED** [review]

File: server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java (hunk 1 above)
Comment (132ikl) [reply server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java:17]: 10s*60s = 600s * 20t = 12000 ticks = 10 mins

**Comment by john01dav** [comment]
Did you verify that it's 20 ticks per second?

**Comment by 132ikl** [comment]
It is

https://github.com/ModdedMinecraftClub/polychat2/blob/master/server/src/main/java/club/moddedminecraft/polychat/server/PolychatServer.java#L38

**Pull request merged**